govcd/vm.go

```diff
@@ -1530,3 +1523,214 @@ func (vm *VM) UpdateVmCpuAndMemoryHotAddAsync(cpuHot, memoryAdd bool) (Task, err
+func addEmptyVmAsyncV10(vapp *VApp, reComposeVAppParams *types.RecomposeVAppParamsForEmptyVm) (Task, error) {
+	err := validateEmptyVmParams(reComposeVAppParams)
+	if err != nil {
+		return Task{}, err
+	}
+	apiEndpoint, _ := url.ParseRequestURI(vapp.VApp.HREF)
+	apiEndpoint.Path += "/action/recomposeVApp"
+
+	reComposeVAppParams.XmlnsVcloud = types.XMLNamespaceVCloud
+	reComposeVAppParams.XmlnsOvf = types.XMLNamespaceOVF
+
+	// Return the task
+	return vapp.client.ExecuteTaskRequestWithApiVersion(apiEndpoint.String(), http.MethodPost,
+		types.MimeRecomposeVappParams, "error instantiating a new VM: %s", reComposeVAppParams,
+		vapp.client.GetSpecificApiVersionOnCondition(">= 33.0", "33.0"))
+}
+
+// addEmptyVmV10 adds an empty VM (without template) to vApp and returns the new created VM or an error.
+func addEmptyVmV10(vapp *VApp, reComposeVAppParams *types.RecomposeVAppParamsForEmptyVm) (*VM, error) {
+	task, err := addEmptyVmAsyncV10(vapp, reComposeVAppParams)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
+
+	err = vapp.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing vapp: %s", err)
+	}
+
+	//vApp Might Not Have Any VMs
+	if vapp.VApp.Children == nil {
+		return nil, ErrorEntityNotFound
+	}
+
+	util.Logger.Printf("[TRACE] Looking for VM: %s", reComposeVAppParams.CreateItem.Name)
+	for _, child := range vapp.VApp.Children.VM {
+
+		util.Logger.Printf("[TRACE] Looking at: %s", child.Name)
+		if child.Name == reComposeVAppParams.CreateItem.Name {
+			return getVMByHrefV10(vapp.client, child.HREF)
+		}
+
+	}
+	util.Logger.Printf("[TRACE] Couldn't find VM: %s", reComposeVAppParams.CreateItem.Name)
+	return nil, ErrorEntityNotFound
+}
+
+// addEmptyVmAsyncV97 adds an empty VM (without template) to the vApp and returns a Task and an error.
+func addEmptyVmAsyncV97(vapp *VApp, reComposeVAppParams *types.RecomposeVAppParamsForEmptyVm) (Task, error) {
+	err := validateEmptyVmParams(reComposeVAppParams)
+	if err != nil {
+		return Task{}, err
+	}
+	apiEndpoint, _ := url.ParseRequestURI(vapp.VApp.HREF)
+	apiEndpoint.Path += "/action/recomposeVApp"
+
+	reComposeVAppParams.XmlnsVcloud = types.XMLNamespaceVCloud
+	reComposeVAppParams.XmlnsOvf = types.XMLNamespaceOVF
+
+	// Return the task
+	return vapp.client.ExecuteTaskRequest(apiEndpoint.String(), http.MethodPost,
+		types.MimeRecomposeVappParams, "error instantiating a new VM: %s", reComposeVAppParams)
+}
+
+// addEmptyVmV97 adds an empty VM (without template) to vApp and returns the new created VM or an error.
+func addEmptyVmV97(vapp *VApp, reComposeVAppParams *types.RecomposeVAppParamsForEmptyVm) (*VM, error) {
+	task, err := addEmptyVmAsyncV97(vapp, reComposeVAppParams)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, err
+	}
+
+	err = vapp.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing vapp: %s", err)
```

Comment:
In all messages we should capitalize `vApp`:
```suggestion
		return nil, fmt.Errorf("error refreshing vApp: %s", err)
```

Reply:
changed everywhere in project